packages/core/src/components/non-ideal-state/nonIdealState.tsx

```diff
@@ -119,21 +105,30 @@ export class NonIdealState extends AbstractPureComponent<NonIdealStateProps> {
                         tabIndex={-1}
                     />
                 </div>
-            );
-        }
-    }
-
-    private maybeRenderText() {
-        const { description, title } = this.props;
-        if (title == null && description == null) {
-            return undefined;
-        } else {
-            return (
+            )
+        );
+    };
+
+    const maybeRenderText = () => {
```

Comment:
Instead of keeping these as inline functions, can we encapsulate the conditional in a couple variables, say:
```tsx
const shouldRenderVisual = icon != null;
const shouldRenderText = title != null || description != null;
```

and then inline them below:
```tsx
<div className={classNames(Classes.NON_IDEAL_STATE, `${Classes.NON_IDEAL_STATE}-${layout}`, className)}>
    {shouldRenderVisual && (
        <div
            className={Classes.NON_IDEAL_STATE_VISUAL}
            style={{ fontSize: `${iconSize}px`, lineHeight: `${iconSize}px` }}
        >
            <Icon
                className={classNames({ [Classes.ICON_MUTED]: iconMuted })}
                icon={icon}
                size={iconSize}
                aria-hidden={true}
                tabIndex={-1}
            />
        </div>
    )}
    {shouldRenderText && (
        <div className={Classes.NON_IDEAL_STATE_TEXT}>
            {title && <H4>{title}</H4>}
            {description && ensureElement(description, "div")}
        </div>
    )}
    {action}
    {children}
</div>
```
I think this would improve the code clarity a bit, and it precludes having to make additional unnecessary function calls.